File: ar-app-android-project/app/src/main/java/com/example/robotkarolar/karollogic/instructions/expressions/Not.java
package com.example.robotkarolar.karollogic.instructions.expressions;

import com.example.robotkarolar.karollogic.instructions.visitors.InstructionVisitor;

public class Not extends Expression {

    private final Expression child;

    public Not(Expression child) {
        this.child = child;
    }

    public Expression getChild() {
        return child;
    }

    @Override
    public void accept(InstructionVisitor visitor) {
        visitor.accept(this);
    }

    @Override
    public boolean accept(ExpressionVisitor visitor) {
        return visitor.accept(this);
    }
}

File: ar-app-android-project/app/src/main/java/com/example/robotkarolar/karollogic/instructions/expressions/Not.kt
package com.example.robotkarolar.karollogic.instructions.expressions

import com.example.robotkarolar.karollogic.instructions.visitors.InstructionVisitor
import kotlinx.parcelize.Parcelize

@Parcelize
class Not(var child: Expression) : Expression() {
    override fun accept(visitor: InstructionVisitor?) {
        visitor!!.accept(this)
    }

    override fun accept(visitor: ExpressionVisitor): Boolean {
        return visitor.accept(this)
    }
}